### PR TITLE
Us6004   email notifications

### PR DESCRIPTION
**Pull request by yujlin2**
Automatic email notifications

### CODE REVIEW

File: src/components/EmailResponses.tsx [hunk 1]
```diff
@@ -7,94 +7,362 @@
...
+	// Handle course ID input change
+	const handleAssignmentIdChange = (e: ChangeEvent<HTMLInputElement>) => {
+		const selectedAssignmentId = Number(e.target.value);
+		// Update the assignmentId state with the selectedAssignmentId
+		setAssignmentId(selectedAssignmentId);
+		setIsAssignmentSelected(true);
+		if (course) {
+			const teacher = course.enrollments.some(
+				(enrollment) =>
+					enrollment.role == "TeacherEnrollment" ||
```
Comment (Stefce02) [inline src/components/EmailResponses.tsx:64]: Change to use env values. Observer and Designer also count as teachers so can do the negation of student/account

File: src/components/RequestForm.tsx [hunk 2]
```diff
@@ -452,6 +459,50 @@ function RequestForm() {
 			console.error("Existing request for same assignmentId");
 			return;
 		}
+
```
Comment (Stefce02) [inline src/components/RequestForm.tsx:462]: Remove if not using

**Review by Dragan14 — APPROVED** [review]
sublime

File: src/hooks/useStaffInCourse.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	// useEffect hook to fetch the users data when the component is mounted or when groupId changes
+	useEffect(() => {
+		if (!domain || !courseId) {
+			return;
+		}
+
+		console.log("useStaffInCourse");
+
+		const teacher = process.env.VITE_teacherEnrollment_roleOne?.replace(
```
Comment (Stefce02) [inline src/hooks/useStaffInCourse.tsx:44]: What about observer and designer enrollments?

**Review by Stefce02 — APPROVED** [review]

**Pull request merged**